hal/src/nRF52840/sleep_hal.cpp

```diff
@@ -265,6 +292,75 @@ static void configRtcWakeupSource(const hal_wakeup_source_base_t* wakeupSources)
     }
 }
 
+static void configLpcompWakeupSource(const hal_wakeup_source_base_t* wakeupSources) {
+    Hal_Pin_Info* halPinMap = HAL_Pin_Map();
+    auto source = wakeupSources;
+    while (source) {
+        if (source->type == HAL_WAKEUP_SOURCE_TYPE_LPCOMP) {
+            auto lpcompWakeup = reinterpret_cast<const hal_wakeup_source_lpcomp_t*>(source);
+            nrf_lpcomp_config_t config = {};
+
+            NVIC_ClearPendingIRQ(COMP_LPCOMP_IRQn);
+
+            // Reference voltage is VDD/16 * N, N ranges from 1 to 15, VDD/16 = 206mV
+            const nrf_lpcomp_ref_t refs[] = {
+                NRF_LPCOMP_REF_SUPPLY_1_16, NRF_LPCOMP_REF_SUPPLY_1_8,
+                NRF_LPCOMP_REF_SUPPLY_3_16, NRF_LPCOMP_REF_SUPPLY_2_8,
+                NRF_LPCOMP_REF_SUPPLY_5_16, NRF_LPCOMP_REF_SUPPLY_3_8,
+                NRF_LPCOMP_REF_SUPPLY_7_16, NRF_LPCOMP_REF_SUPPLY_4_8,
+                NRF_LPCOMP_REF_SUPPLY_9_16, NRF_LPCOMP_REF_SUPPLY_5_8,
+                NRF_LPCOMP_REF_SUPPLY_11_16, NRF_LPCOMP_REF_SUPPLY_6_8,
+                NRF_LPCOMP_REF_SUPPLY_13_16, NRF_LPCOMP_REF_SUPPLY_7_8,
+                NRF_LPCOMP_REF_SUPPLY_15_16
+            };
+            uint8_t n = lpcompWakeup->voltage / 103; // Half of 206mV
+            n = (n + (n % 2)) >> 1;
```

Comment:
Why so much complexity? Each of the values is exactly 1/16 or 103mV apart. It's fine to just perform integer division with any rounding flavor (which might be perhaps different depending on which direction the trigger goes).
Floor: just divide by 103.
Ceil: `(x + (103 - 1)) / 103`
Math round: `(x + (103 / 2)) / 103`